lang/csharp/src/apache/main/Reflect/ClassCache.cs

```diff
@@ -29,6 +31,8 @@ public class ClassCache
     {
         private static ConcurrentBag<IAvroFieldConverter> _defaultConverters = new ConcurrentBag<IAvroFieldConverter>();
 
+        private static ConcurrentDictionary<Tuple<Type, Type, bool>, IAvroFieldConverter> _typeDefaultConverters = new ConcurrentDictionary<Tuple<Type, Type, bool>, IAvroFieldConverter>();
```

Comment:
This key is MEAN ... soo freaking many types

Reply:
I chose Tuple here simply so I didn't have to write my own IEquatable/HashCode code.  Tuple takes care of that for me, and Type, bool both have their own IEquatable, so it works.